### PR TITLE
New site navbar, lang picking, refactors

### PR DESCRIPTION
**Pull request by WiIIiam278**
* Add global navbar
* Add site language picker dropdown
* Add patch language picker for international translations
* Refactor social links
* Refactor global layout structure a bit
* Link to our YouTube
* Once again fix shiki ARM syntax highlighting
* Updates nuxt and nuxt-content to latest versions

### CODE REVIEW

File: pages/index.vue
Comment (jonko0493) [inline pages/index.vue:44]: hmm we should look at pulling these strings from the locale strings if possible, probably. I don't think I did that last time but it would be good.

**Review by jonko0493 — APPROVED** [review]
Looks great! I really appreciate that you used lang names for the picker rather than flags lol. Do you want to wait until I merge my blog post to merge? This was done faster than I expected haha.

**Comment by WiIIiam278** [comment]
> Looks great! I really appreciate that you used lang names for the picker rather than flags lol. Do you want to wait until I merge my blog post to merge? This was done faster than I expected haha.

yeah guess we can merge this before then

**Pull request merged**
